src/inmanta/data/__init__.py

```diff
@@ -2298,13 +2633,148 @@ async def get_resources_for_version_raw(
         return resources
 
     @classmethod
-    async def get_latest_version(cls, environment: uuid.UUID, resource_id: ResourceIdStr) -> Optional["Resource"]:
+    async def get_latest_version(cls, environment: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional["Resource"]:
         resources = await cls.get_list(
             order_by_column="model", order="DESC", limit=1, environment=environment, resource_id=resource_id
         )
         if len(resources) > 0:
             return resources[0]
 
+    @classmethod
+    def _get_released_resources_base_query(cls):
+        return (
+            f"FROM ( "
+            f"SELECT r.*, row_number() OVER (PARTITION BY r.resource_id ORDER BY r.model DESC) as row_number "
```

Comment:
I would have expected `DISTINCT ON`  (never seen this before), but I have no idea what is best.